AppCenter/AppCenter/Internals/Sender/MSSenderCall.m

```diff
@@ -81,10 +82,20 @@ - (void)sender:(id<MSSender>)sender
     MSLogInfo([MSAppCenter logTag], logMessage);
     [sender suspend];
   }
-
-  // Retry.
-  else if ([MSSenderUtil isRecoverableError:statusCode] && ![self hasReachedMaxRetries]) {
-    [self startRetryTimerWithStatusCode:statusCode];
+  else if ([MSSenderUtil isRecoverableError:statusCode]) {
+    
+    // Reset the retry count, will retry once app goes to foreground again.
+    if ([self hasReachedMaxRetries]) {
+      [self resetRetry];
+      NSString *logMessage = @"Received recoverable error but reached max. number of retries.";
+      MSLogInfo([MSAppCenter logTag], logMessage);
+      [sender suspend];
```

Comment:
I think we should do more than this. I think we should remove pending batch ID and pending call ID, disable it instead of just suspending by calling `[self setEnabled:NO addDeleteDataOnDisabled: NO]`